src/main/scala/daisy/utils/CPrinter.scala

```diff
@@ -430,6 +430,7 @@ class CPrinter(buffer: Appendable, ctx: Context) extends CodePrinter(buffer) {
         }
 
         sb.append(") {")
+        sb.append("\n#pragma HLS PIPELINE\n#pragma HLS ARRAY_PARTITION variable=_result complete")
         nl(lvl+1)
```

Comment:
Where are the other changes in this file? I assume you had to edit for the nested code part as well.

Reply:
The nested code part is in separate file CPrinterOptimized.scala